accord-core/src/main/java/accord/messages/ReadTxnData.java

```diff
@@ -231,8 +231,14 @@ protected synchronized void readComplete(CommandStore commandStore, @Nullable Da
     }
 
     @Override
-    protected void reply(@Nullable Ranges unavailable, @Nullable Data data)
+    protected void reply(@Nullable Ranges unavailable, @Nullable Data data, @Nullable Throwable fail)
     {
+        if (fail != null)
+        {
+            node.reply(replyTo, replyContext, null, fail);
```

Comment:
```suggestion
            state = State.RETURNED;
            node.reply(replyTo, replyContext, null, fail);
```

Reply:
Shouldn't this also have the same illegal state logic as well? Maybe the fail handling should be in the switch below and handle obsolete as well.